cmd/init.go

```diff
@@ -576,3 +577,42 @@ func parseProjectParm(projectParm string, config *RootCommandConfig) (string, st
 
 	return "", "", errors.New("malformed project parameter - something unusual happened")
 }
+
+func defaultProjectName(config *RootCommandConfig) string {
+	projectName, _ := getProjectName(config)
+
+	if projectName == "" || projectName == "my-project" {
+
+		projectDirPath, err := os.Getwd()
+
+		if err != nil {
+			Error.log(err)
+			os.Exit(1)
+		}
+
+		indexPathSeparator := strings.LastIndex(projectDirPath, string(os.PathSeparator))
+
+		projectName := projectDirPath[indexPathSeparator+1:]
+
+		projectName = strings.ToLower(filepath.Base(projectName))
+		match, _ := regexp.MatchString("[a-z]([-a-z0-9]*[a-z0-9])?", projectName)
+		if match {
+			return projectName
+		}
+		projectName = "appsody-" + strings.ToLower(filepath.Base(projectName)) + "-app"
+		reg, err := regexp.Compile("[^a-z0-9]+")
+		if err != nil {
+			return ""
+		}
+		projectName = reg.ReplaceAllString(projectName, "-")
+
+		match, _ = regexp.MatchString("[a-z]([-a-z0-9]*[a-z0-9])?", projectName)
+
+		if match {
+			return projectName
+		}
+
+		return "my-project"
+	}
+	return projectName
+}
```

Comment:
This logic (or a small variation thereof) is repeated three times - perhaps we can check this in as it is now and refactor later? @kylegc what do you think...?

Reply:
I have refactored it